api/pkg/artifact/artifact.go

```diff
@@ -9,25 +9,33 @@ import (
 )
 
 type gcsClient struct {
-	API    *storage.Client
-	Config Config
+	API *storage.Client
 }
+
+type nopArtifact struct{}
```

Comment:
Okay hang on, can we standardise the naming convention for the `gcsClient` and `nopArtifact`? 😅 Same thing for the methods `NewGcsArtifactService` and `NewNopArtifactService` that initialise them?

Since they are both implementations of the `artifact.Service` interface, can we have something like,
`gcsArtifactClient` and `nopArtifactClient`, and same thing for the methods, `NewGcsArtifactClient` and `NewNopArtifactClient`?  I'm okay with any names as long as they are all consistent 😅